packages/swapper/src/swappers/ThorchainSwapper/endpoints.ts

```diff
@@ -328,13 +329,36 @@ export const thorchainApi: SwapperApi = {
 
   checkTradeStatus: async ({
     txHash,
+    chainId,
     config,
   }): Promise<{
     status: TxStatus
     buyTxHash: string | undefined
     message: string | undefined
   }> => {
     try {
+      const safeTransactionInfo = await fetchSafeTransactionInfo({
+        chainId,
+        maybeSafeTxHash: txHash,
+      })
+      const { isSafeTxHash, transaction } = safeTransactionInfo
+
+      // No buyTxHash handling is correct - we mutate with the actual on-chain transaction, meaning the regular flow then takes over
+      if (
+        isSafeTxHash &&
+        transaction?.confirmations &&
+        Number(transaction.confirmations.length) < transaction.confirmationsRequired
+      ) {
+        return {
+          status: TxStatus.Pending,
+          message: `SAFE proposal submitted. ${transaction.confirmations.length} out of ${transaction.confirmationsRequired} signed.`,
+          buyTxHash: undefined,
+        }
+      } else if (transaction?.transactionHash) {
+        // Mutate with  the actual on-chain transaction work and let things work as-is
```

Comment:
```suggestion
        // Mutate with the actual on-chain transaction hash instead of using the safe transaction hash and continue with the normal status detection flow
```

Reply:
:brainlet: https://github.com/shapeshift/web/pull/7506/commits/3b137148d3fad0b1814411f3a3ad0a77c1cd07bc